pangolin/4.1.2/Dockerfile

```diff
@@ -0,0 +1,111 @@
+# create the conda environment using modified environment.yml
+RUN sed -i "s|usher.*|usher=${USHER_VER}|" /pangolin/environment.yml && \
+ sed -i "s|scorpio.git|scorpio.git@${SCORPIO_VER}|" /pangolin/environment.yml && \
+ sed -i "s|pangolin-data.git|pangolin-data.git@${PANGOLIN_DATA_VER}|" /pangolin/environment.yml && \
+ sed -i "s|constellations.git|constellations.git@${CONSTELLATIONS_VER}|" /pangolin/environment.yml && \
+ micromamba create -n pangolin -y -f /pangolin/environment.yml
+
+# so that mamba/conda env is active when running below commands
+ENV ENV_NAME="pangolin"
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+WORKDIR /pangolin
+
+# run pip install step; download optional pre-computed assignment hashes for UShER (useful for running on large batches of samples)
+# best to skip using the assigment-cache if running on one sample for speed
+# print versions
+RUN pip install . && \
+ pangolin --add-assignment-cache && \
+ micromamba clean -a -y && \
+ mkdir /data && \
+ pangolin --all-versions && \
+ usher --version
+
+WORKDIR /data
+
+# hardcode pangolin executable into the PATH variable
+ENV PATH="$PATH:/opt/conda/envs/pangolin/bin/"
+
+# new base for testing
+FROM app as test
+
+# so that mamba/conda env is active when running below commands
+ENV ENV_NAME="pangolin"
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
+
+# install unzip for unzipping zip archive from NCBI
+RUN apt-get update && apt-get install -y --no-install-recommends unzip
+
+# install ncbi datasets tool (pre-compiled binary); place in $PATH
+RUN wget https://ftp.ncbi.nlm.nih.gov/pub/datasets/command-line/LATEST/linux-amd64/datasets && \
+ chmod +x datasets && \
+ mv -v datasets /usr/local/bin
+
+# download assembly for a BA.1 from Florida (https://www.ncbi.nlm.nih.gov/biosample?term=SAMN29506515 and https://www.ncbi.nlm.nih.gov/nuccore/ON924087)
+# run pangolin in usher analysis mode
+RUN datasets download virus genome accession ON924087.1 --filename ON924087.1.zip && \
+ unzip ON924087.1.zip && rm ON924087.1.zip && \
+ mv -v ncbi_dataset/data/genomic.fna ncbi_dataset/data/ON924087.1.genomic.fna && \
+ pangolin ncbi_dataset/data/ON924087.1.genomic.fna --analysis-mode usher -o ON924087.1-usher && \
+ cat ON924087.1-usher/lineage_report.csv
+
+# test on test sequences supplied with Pangolin code
+RUN pangolin /pangolin/pangolin/test/test_seqs.fasta --analysis-mode usher -o /data/test_seqs-output-pusher && \
+ cat /data/test_seqs-output-pusher/lineage_report.csv 
+
+# test functionality of assignment-cache option
+RUN pangolin --use-assignment-cache /pangolin/pangolin/test/test_seqs.fasta
+
+# download B.1.1.7 genome from Utah
+ADD https://raw.githubusercontent.com/StaPH-B/docker-builds/master/tests/SARS-CoV-2/SRR13957123.consensus.fa /test-data/SRR13957123.consensus.fa
+
+# test on a B.1.1.7 genome
+RUN pangolin /test-data/SRR13957123.consensus.fa --analysis-mode usher -o /test-data/SRR13957123-pusher && \
+ cat /test-data/SRR13957123-pusher/lineage_report.csv
```

Comment:
Since these don't require datasets installed, can you have these run before datasets and unzip are added?

Reply:
sure thing, I will change the order